src/safe_op.hpp

```diff
@@ -331,4 +332,103 @@ namespace Safe
         return num < 0 ? -num : num;
     }
 
+    namespace Internal
+    {
+        template <typename from, typename to, typename = void>
+        struct is_safely_convertible : std::false_type
+        {
+            static_assert(std::is_integral<from>::value&& std::is_integral<to>::value,
+                          "from and to must both be integral types");
+        };
+
+        template <typename from, typename to>
+        struct is_safely_convertible<
+            from, to,
+            typename std::enable_if<((std::numeric_limits<from>::max() <= std::numeric_limits<to>::max()) &&
+                                     (std::numeric_limits<from>::min() >= std::numeric_limits<to>::min()))>::type>
+            : std::true_type
+        {
+        };
+
+        template <typename T, typename U, typename = void>
+        struct have_same_signedness : std::false_type
+        {
+            static_assert(std::is_integral<T>::value&& std::is_integral<U>::value,
```

Comment:
This confused me at first, because I read `value&&` as an rvalue reference type. I would recommend adding a space to make it clear that this is just the regular Boolean `&&` operator:

```
std::is_integral<from>::value && std::is_integral<to>::value
```

(Also above in `is_safely_convertible`.)

Reply:
I blame clang format for this one :wink: